gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/LoadConfiguration.java

```diff
@@ -0,0 +1,381 @@
+     *     Create Disposition</a>
+     */
+    public Builder createDisposition(CreateDisposition createDisposition) {
+      this.createDisposition = createDisposition;
+      return this;
+    }
+
+    /**
+     * Sets the action that should occur if the destination table already exists.
+     *
+     * @see <a href="https://cloud.google.com/bigquery/docs/reference/v2/jobs#configuration.load.writeDisposition">
+     *     Write Disposition</a>
+     */
+    public Builder writeDisposition(WriteDisposition writeDisposition) {
+      this.writeDisposition = writeDisposition;
+      return this;
+    }
+
+    /**
+     * Sets the source format, and possibly some parsing options, of the external data. Supported
+     * formats are {@code CSV}, {@code NEWLINE_DELIMITED_JSON} and {@code DATASTORE_BACKUP}. If not
+     * specified, {@code CSV} format is assumed.
+     *
+     * <a href="https://cloud.google.com/bigquery/docs/reference/v2/tables#externalDataConfiguration.sourceFormat">
+     *     Source Format</a>
+     */
+    public Builder formatOptions(FormatOptions formatOptions) {
+      this.formatOptions = formatOptions;
+      return this;
+    }
+
+    /**
+     * Sets the maximum number of bad records that BigQuery can ignore when running the job. If the
+     * number of bad records exceeds this value, an invalid error is returned in the job result.
+     * By default no bad record is ignored.
+     */
+    public Builder maxBadRecords(Integer maxBadRecords) {
+      this.maxBadRecords = maxBadRecords;
+      return this;
+    }
+
+    /**
+     * Sets the schema for the destination table. The schema can be omitted if the destination table
+     * already exists, or if you're loading data from Google Cloud Datastore.
+     */
+    public Builder schema(Schema schema) {
+      this.schema = schema;
+      return this;
+    }
+
+    /**
+     * Sets whether BigQuery should allow extra values that are not represented in the table schema.
+     * If {@code true}, the extra values are ignored. If {@code false}, records with extra columns
+     * are treated as bad records, and if there are too many bad records, an invalid error is
+     * returned in the job result. By default unknown values are not allowed.
+     */
+    public Builder ignoreUnknownValues(Boolean ignoreUnknownValues) {
+      this.ignoreUnknownValues = ignoreUnknownValues;
+      return this;
+    }
+
+    /**
+     * Sets which entity properties to load into BigQuery from a Cloud Datastore backup. This field
+     * is only used if the source format is set to {@code DATASTORE_BACKUP}. Property names are case
+     * sensitive and must be top-level properties. If no properties are specified, BigQuery loads
+     * all properties. If any named property isn't found in the Cloud Datastore backup, an invalid
+     * error is returned in the job result.
+     */
+    public Builder projectionFields(List<String> projectionFields) {
```

Comment:
shouldn't it be part of the DATASTORE_BACKUP FormatOptions? I mean via a subclasses, similar to CSV?